src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -453,6 +453,64 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or FALSE on failure.
+     */
+    public function tempNam($dir, $prefix)
```

Comment:
Happy to add to symfony-docs if this feature is wanted.

Without a stream aware version of tempnam it is very difficult to write to a stream atomically. The path of least resistance becomes writing to the stream non-atomically using file_put_contents, circumventing the benefit of dumpFile. See #10018.